release v1.0.2 (#1080)
pingcap/tidb-operator: CHANGELOG.md

# TiDB Operator v1.0.2 Release Notes

## v1.0.2 What's New

### Action Required

> The AWS Terraform script uses auto-scaling-group for all components (pd/tikv/tidb/monitor), when an ec2 instance fails the health check, the ec2 instance will be replaced. This is helpful for stateless applications or applications using EBS volumes to store data.

Reviewer: ```suggestion
The AWS Terraform script uses auto-scaling-group for all components (PD/TiKV/TiDB/monitor). When an ec2 instance fails the health check, the instance will be replaced. This is helpful for those applications that are stateless or use EBS volumes to store data.
```

> But TiKV pod uses instance store to store its data. When the instance is replaced, all the data on the instance store will be lost. TiKV has to resync all data to the newly added instance. Though TiDB is a distributed database and can work when a node fails, the cost to resync data is quite big if the dataset is large. Besides, the ec2 instance may recover to a healthy status by rebooting.

Reviewer: ```suggestion
But a TiKV Pod uses instance store to store its data. When an instance is replaced, all the data on its store will be lost. TiKV has to resync all data to the newly added instance. Though TiDB is a distributed database and can work when a node fails, resyncing data can cost much if the dataset is large. Besides, the ec2 instance may be recovered to a healthy state by rebooting.
```

> So we disable the auto-scaling-group's replace behavior in `v1.0.2`.

Reviewer: ```suggestion
So we disabled the auto-scaling-group's replacing behavior in `v1.0.2`.
```

Auto-scaling-group scaling process can also be suspended according to its [documentation](https://docs.aws.amazon.com/autoscaling/ec2/userguide/as-suspend-resume-processes.html) if you are using `v1.0.1` or prior versions.

### Improvements

> - Migrating statefulsets apiVersion to `app/v1`

Reviewer 2: This is a bugfix to fix compatiblity with k8s 1.16 and above versions.

> - Suspend ReplaceUnhealthy process for AWS tikv auto-scaling-group

Reviewer: ```suggestion
- Suspend ReplaceUnhealthy process for AWS TiKV auto-scaling-group
```

- Add a new VM manager `qm` in stability test
- Add `tikv.maxFailoverCount` limit to TiKV

> - Set the default `externalTrafficPolicy` to be `Local` for TiDB service in aws/gcp/aliyun

Reviewer: ```suggestion
- Set the default `externalTrafficPolicy` to be `Local` for TiDB service in AWS/GCP/Aliyun
```

> - Add provider and module versions for aws

Reviewer: ```suggestion
- Add provider and module versions for AWS
```

### Bug fixes

> - Migrating statefulsets apiVersion to `app/v1` which fixes compatiblity with k8s 1.16 and above versions

Reviewer: ```suggestion
- Migrate statefulsets apiVersion to `app/v1` which fixes compatibility with Kubernetes 1.16 and above versions
```

> - Fix that the create_tidb_cluster_release variable in AWS terraform script does not work

Reviewer 2: ```suggestion
- Fix that the `create_tidb_cluster_release` variable in AWS terraform script does not work
```

> - Fix that the `create_tidb_cluster_release` variable in AWS terraform script does not work

Reviewer: ```suggestion
- Fix the issue that the `create_tidb_cluster_release` variable in AWS Terraform script does not work
```

> - Fix compatibility issues for ([#1012](https://github.com/pingcap/tidb-operator/pull/1012)): add `v1beta1` to statefulset apiVersions

Reviewer: ```suggestion
- Fix compatibility issues by adding `v1beta1` to statefulset apiVersions
```

> - Fix the issue that TiDB Loadbalancer is empty in terraform output

Reviewer: ```suggestion
- Fix the issue that TiDB Loadbalancer is empty in Terraform output
```

> - Fix a compatibility issue on TiKV maxFailoverCount

Reviewer: ```suggestion
- Fix a compatibility issue of TiKV `maxFailoverCount`
```

> - Fix provider versions constraint issues for GCP and Aliyun

Reviewer 2: ```suggestion
- Fix Terraform providers version constraint issues for GCP and Aliyun
```

- Fix values file customization for tidb-operator on Aliyun

> - Fix tidb-operator crash when user modify sts upgrade strategy improperly

Reviewer 2: ```suggestion
- Fix tidb-operator crash when users modify statefulset upgrade strategy improperly
```

- Fix drainer misconfiguration

## Detailed Bug Fixes and Changes

> - Fix that the create_tidb_cluster_release variable in AWS terraform script does not work ([#1062](https://github.com/pingcap/tidb-operator/pull/1062))

Reviewer 2: ```suggestion
- Fix that the `create_tidb_cluster_release` variable in AWS terraform script does not work ([#1062](https://github.com/pingcap/tidb-operator/pull/1062))
```

> - Fix that the create_tidb_cluster_release variable in AWS terraform script does not work ([#1062](https://github.com/pingcap/tidb-operator/pull/1062))

Reviewer: ```suggestion
- Fix the issue that the `create_tidb_cluster_release` variable in AWS Terraform script does not work ([#1062](https://github.com/pingcap/tidb-operator/pull/1062))
```

- Fix compatibility issues for ([#1012](https://github.com/pingcap/tidb-operator/pull/1012)): add `v1beta1` to statefulset apiVersions ([#1054](https://github.com/pingcap/tidb-operator/pull/1054))
- Enable ConfigMapRollout by default in stability test ([#1036](https://github.com/pingcap/tidb-operator/pull/1036))

> - Fix the issue that TiDB Loadbalancer is empty in terraform output ([#1045](https://github.com/pingcap/tidb-operator/pull/1045))

Reviewer: ```suggestion
- Fix the issue that TiDB Loadbalancer is empty in Terraform output ([#1045](https://github.com/pingcap/tidb-operator/pull/1045))
```

> - Migrating statefulsets apiVersion to `app/v1` which fixes compatiblity with k8s 1.16 and above versions ([#1012](https://github.com/pingcap/tidb-operator/pull/1012))

Reviewer: ```suggestion
- Migrate statefulsets apiVersion to `app/v1` which fixes compatibility with Kubernetes 1.16 and above versions ([#1012](https://github.com/pingcap/tidb-operator/pull/1012))
```

> - Only expect upgrade complete when rollback bad configuration in stability test ([#1030](https://github.com/pingcap/tidb-operator/pull/1030))

Reviewer: ```suggestion
- Only expect upgrade complete when rollback bad configuration in stability test ([#1030](https://github.com/pingcap/tidb-operator/pull/1030))
```
Please explain this sentence in Chinese. It is confusing me.

Author: @aylei PTAL

Reviewer 3: ”在 e2e 测试中回滚错误配置时，（测试用例）只期待（TiDB集群）升级完成“

Reviewer: Thanks~ I edited my suggested change accordingly, PTAL @weekface

> - Suspend ReplaceUnhealthy process for AWS tikv auto-scaling-group ([#1014](https://github.com/pingcap/tidb-operator/pull/1014))

Reviewer: ```suggestion
- Suspend ReplaceUnhealthy process for AWS TiKV auto-scaling-group ([#1014](https://github.com/pingcap/tidb-operator/pull/1014))
```

- Add a new VM manager `qm` in stability test ([#896](https://github.com/pingcap/tidb-operator/pull/896))
- Fix provider versions constraint issues for GCP and Aliyun ([#959](https://github.com/pingcap/tidb-operator/pull/959))

> - Fix values file customization for tidb-operator on aliyun ([#971](https://github.com/pingcap/tidb-operator/pull/971))

Reviewer: ```suggestion
- Fix values file customization for tidb-operator on Aliyun ([#971](https://github.com/pingcap/tidb-operator/pull/971))
```

> - Fix a compatibility issue on TiKV `tikv.maxFailoverCount` ([#977](https://github.com/pingcap/tidb-operator/pull/977))

Reviewer: ```suggestion
- Fix a compatibility issue of TiKV `tikv.maxFailoverCount` ([#977](https://github.com/pingcap/tidb-operator/pull/977))
```

- Add `tikv.maxFailoverCount` limit to TiKV ([#965](https://github.com/pingcap/tidb-operator/pull/965))

> - Fix tidb-operator crash when user modify sts upgrade strategy improperly ([#912](https://github.com/pingcap/tidb-operator/pull/912))

Reviewer: ```suggestion
- Fix tidb-operator crash when users modify statefulset upgrade strategy improperly ([#912](https://github.com/pingcap/tidb-operator/pull/912))
```

> - Set the default `externalTrafficPolicy` to be `Local` for TiDB service in aws/gcp/aliyun ([#947](https://github.com/pingcap/tidb-operator/pull/947))

Reviewer: ```suggestion
- Set the default `externalTrafficPolicy` to be `Local` for TiDB service in AWS/GCP/Aliyun ([#947](https://github.com/pingcap/tidb-operator/pull/947))
```

> - Add note about setting pv reclaim policy to retain ([#911](https://github.com/pingcap/tidb-operator/pull/911))

Reviewer: ```suggestion
- Add note about setting PV reclaim policy to retain ([#911](https://github.com/pingcap/tidb-operator/pull/911))
```

- Fix drainer misconfiguration ([#939](https://github.com/pingcap/tidb-operator/pull/939))

> - Add provider and module versions for aws ([#926](https://github.com/pingcap/tidb-operator/pull/926))

Reviewer: ```suggestion
- Add provider and module versions for AWS ([#926](https://github.com/pingcap/tidb-operator/pull/926))
```

But a TiKV Pod uses instance store to store its data. When an instance is replaced, all the data on its store will be lost. TiKV has to resync all data to the newly added instance. Though TiDB is a distributed database and can work when a node fails, resyncing data can cost much if the dataset is large. Besides, the ec2 instance may be recovered to a healthy state by rebooting.

So we disabled the auto-scaling-group's replacing behavior in `v1.0.2`.

Auto-scaling-group scaling process can also be suspended according to its [documentation](https://docs.aws.amazon.com/autoscaling/ec2/userguide/as-suspend-resume-processes.html) if you are using `v1.0.1` or prior versions.

### Improvements

- Suspend ReplaceUnhealthy process for AWS TiKV auto-scaling-group
- Add a new VM manager `qm` in stability test
- Add `tikv.maxFailoverCount` limit to TiKV
- Set the default `externalTrafficPolicy` to be `Local` for TiDB service in AWS/GCP/Aliyun
- Add provider and module versions for AWS

### Bug fixes

> - Fix tkctl version does not work when the release name is un-wanted

Reviewer: ```suggestion
- Fix the issue that tkctl version does not work when the release name is un-wanted
```

- Migrate statefulsets apiVersion to `app/v1` which fixes compatibility with Kubernetes 1.16 and above versions
- Fix the issue that the `create_tidb_cluster_release` variable in AWS Terraform script does not work
- Fix compatibility issues by adding `v1beta1` to statefulset apiVersions
- Fix the issue that TiDB Loadbalancer is empty in Terraform output
- Fix a compatibility issue of TiKV `maxFailoverCount`
- Fix Terraform providers version constraint issues for GCP and Aliyun
- Fix values file customization for tidb-operator on Aliyun
- Fix tidb-operator crash when users modify statefulset upgrade strategy improperly
- Fix drainer misconfiguration

## Detailed Bug Fixes and Changes

> - Fix tkctl version does not work when the release name is un-wanted ([#1065](https://github.com/pingcap/tidb-operator/pull/1065))

Reviewer: ```suggestion
- Fix the issue that tkctl version does not work when the release name is un-wanted ([#1065](https://github.com/pingcap/tidb-operator/pull/1065))
```

> - Fix that the `create_tidb_cluster_release` variable in AWS terraform script does not work ([#1062](https://github.com/pingcap/tidb-operator/pull/1062))

Reviewer: ```suggestion
- Fix the issue that the `create_tidb_cluster_release` variable in AWS terraform script does not work ([#1062](https://github.com/pingcap/tidb-operator/pull/1062))
```
